emacs/caml.el

```diff
@@ -866,10 +866,10 @@ possible."
  (if (eq major-mode 'caml-mode)
      (let (skip bol beg end)
        (save-excursion
-         (set-buffer
-          (if (boundp 'compilation-last-buffer)
-              compilation-last-buffer   ;Emacs 19
-            "*compilation*"))           ;Emacs 18
+         (with-current-buffer
+             (if (boundp 'compilation-last-buffer)
+                 compilation-last-buffer   ;Emacs 19
+               "*compilation*"))           ;Emacs 18
```

Comment:
I know little about Emacs-lisp in general but this change looks very odd to me. `with-current-buffer` takes a body and runs it with its first argument as the current buffer, so I would expect that the rest of the code below should be the body. Otherwise here the body is empty, so this command should have no effect, while `set-buffer` changed the current buffer for the rest of the command code below.